mathics/eval/arithmetic.py

```diff
@@ -34,22 +35,39 @@
 RationalMOneHalf = Rational(-1, 2)
 
 
-# @lru_cache(maxsize=4096)
 def call_mpmath(
-    mpmath_function: Callable, mpmath_args: tuple, prec: Optional[int] = None
-):
+    mpmath_function: Callable, mpmath_args: tuple, precision: Optional[int] = None
+) -> Optional[BaseElement]:
+    """A wrapper around ``call_mpmath_with_precision()`` which is defined below
+    below.
+
+    The parameter ``precision`` specifies the number bit of precsion
+    of the result.
+
+    If the parameter ``precision`` is not specified, IEEE-754 "double
+    precision" or (about 53 bits) is used.
+    """
+    if precision is None:
+        precision = FP_MANTISA_BINARY_DIGITS
+    return call_mpmath_with_precision(mpmath_function, mpmath_args, precision)
```

Comment:
Maybe this is silly, but can't this function call slow down the code as it is used in a lot of places?

Reply:
I remember I tried this, and I didn't see any improvement in performance, but neither a noticeable overhead. The problem is that in general, we do not call this function over the same values many times.

Reply:
> Maybe this is silly, but can't this function call slow down the code as it is used in a lot of places?

In fact right now we can remove it since the single caller doesn't call this at all any more. 

> I remember I tried this, and I didn't see any improvement in performance, but neither a noticeable overhead. 

When this happens, it would be helpful to _communicate_ this, and then I can focus on things other than this.

> The problem is that in general, we do not call this function over the same values many times.

The situation right now is that we only call this from one place, gamma. And that's because mpmath seems to provide gamma while sympy doesn't  Will that always be the case? I don't know.

Unless I ran this over a large corpus of WMA code designed for benchmarking, I would be hesitant to offer generalizations based on insufficient data. 

At any rate, what to do?  The PR has a number of other small fixes. So I can: 

* So I can pull the changes to call_mpmath and leave the other changes. 
* Another possibility is to require `call_mpmath()` to pass the precision and not make it optional.   So basically `call_mpmath_with_precision` basically renamed back to  `call_mpmath` but there is no longer a default value. 
* Something else

Let me know which way to go.

Reply:
> > I remember I tried this, and I didn't see any improvement in performance, but neither a noticeable overhead.
> 
> When this happens, it would be helpful to _communicate_ this, and then I can focus on things other than this.
> 

Yes, I planned to do that, but I didn't finish all the tests.

> > The problem is that in general, we do not call this function over the same values many times.
> 
> The situation right now is that we only call this from one place, gamma. And that's because mpmath seems to provide gamma while sympy doesn't Will there always only be a single caller? I don't know.
> 

I guess we can keep both callers, and at some point see where it is useful to specialize it.

> Unless I ran this over a large corpus of WMA code designed for benchmarking, I would be hesitant to offer generalizations based on insufficient data.
> 

This is the reason why I didn't report this. The argument about the frequency when the cache is used is just about my intuition on how I use WMA. Maybe in some context, it would be useful to call BesselJ over an array of data with many repeated values, and then it could make more sense. Also, I think that this depends on the particular function.


> At any rate, what to do? The PR has a number of other small fixes. So I can:
> 
>     * Pull the changes to call_mpmath and leave the other changes.
> 
>     * Another possibility is to require `call_mpmath()` to pass the precision and not make it optional.   So basically `call_mpmath_with_precision` basically is renamed back to  `call_mpmath` but there is no longer a default value.
> 
>     * Something else
> 
> 
> Let me know which way to go.

I think that there is not an appreciable overhead, this is fine as it is, and it is easy to remove. I just mentioned this because of the Thiago's comment.